addons/base_iban/models/base_iban.py

```
@@ -0,0 +1,190 @@
+    _inherit = "res.partner.bank"
+
+    @api.model
+    def create(self, vals):
+        #overwrite to format the iban number correctly
+        if (vals.get('state', False) == 'iban') and vals.get('acc_number', False):
+            vals['acc_number'] = _format_iban(vals['acc_number'])
+            vals['acc_number'] = _pretty_iban(vals['acc_number'])
+        return super(ResPartnerBank, self).create(vals)
+
+    @api.multi
+    def write(self, vals):
+        #overwrite to format the iban number correctly
+        if (vals.get('state', False) == 'iban') and vals.get('acc_number', False):
+            vals['acc_number'] = _format_iban(vals['acc_number'])
+            vals['acc_number'] = _pretty_iban(vals['acc_number'])
+        return super(ResPartnerBank, self).write(vals)
+
+    def is_iban_valid(self, iban):
+        """ Check if IBAN is valid or not
+            @param iban: IBAN as string
+            @return: True if IBAN is valid, False otherwise
+        """
+        if not iban:
+            return False
+        iban = _format_iban(iban).lower()
+        if iban[:2] in _ref_iban and len(iban) != len(_format_iban(_ref_iban[iban[:2]])):
+            return False
+        #the four first digits have to be shifted to the end
+        iban = iban[4:] + iban[:4]
+        #letters have to be transformed into numbers (a = 10, b = 11, ...)
+        iban2 = ""
+        for char in iban:
+            if char.isalpha():
+                iban2 += str(ord(char) - 87)
+            else:
+                iban2 += char
+        #iban is correct if modulo 97 == 1
+        return int(iban2) % 97 == 1
+
+    @api.one
+    @api.constrains('acc_number', 'state')
+    def check_iban(self):
+        '''
+        Check the IBAN number
+        '''
+        for bank_acc in self:
+            if bank_acc.state != 'iban':
+                continue
+            if not self.is_iban_valid(bank_acc.acc_number):
+                return False
+        return True
```

raise warning instead of return True False.